### ISSUE TITLE
Riding tied to level of your character

### ISSUE THREAD

**Issue by markvetx** [azerothcore/azerothcore-wotlk#6563]
**CURRENT BEHAVIOUR:**

If you learn riding, it will give the appropriate skill level (75/75, 150/150 etc.)

But, if you re-log, the riding skill points will tie to the level of your character.

**EXPECTED BLIZZLIKE BEHAVIOUR:**

Riding must not be tied to character's level.

**SOURCE:**

**STEPS TO REPRODUCE THE PROBLEM:**

1. Create a char.
2. Learn riding, `.learn 33391` for example.
3. Re-log.
4. Check skill tab.

**EXTRA NOTES:**

**AC HASH/COMMIT:**
 
edf3df5

**OPERATING SYSTEM:**

Arch Linux 5.12

**MODULES:**

None

**CUSTOMIZATIONS:**

**Comment by Hacaw** [comment]
@markvetx I tried to follow your steps, but I think you should add more.

What commands did you use to level up/down, because if you used `character level x` or `level x` indeed I can confirm your report.

As per https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130065/GM+Commands for the `character level` and `level up` command it states that :

> All stats and dependent values recalculated.

I think this might be an issue with the level up commands.

**Comment by markvetx** [comment]
@Hacaw Yes, I did level up my characters with `.lev xx`, but, the same issue is happening on my "non-boosted by command" characters.

One of my characters had 75/75 riding (rank 1), but after some project updates (after 1be561e, I believe), I mentioned this problem.

**Comment by Onidsouza** [comment]
@markvetx It seems that the changes made in project update [1be561e](https://github.com/azerothcore/azerothcore-wotlk/commit/1be561e03b56dc396270335886e59eddad9fa0c6) had even more extra unforeseen consequences. I've checked the changes made to GetSkillRangeType, and at least for me lots of the sql tables that should populate the stores are actually empty in the database (skilllineability_dbc, skillline_dbc, skillraceclassinfo_dbc). 

Moreover, checking the code at DBCStores.cpp, sSkillTiersStore never seems to be actually properly initialized, so my guess is that it is running empty.  Meanwhile, adding the following code to the switch case in line 8038 (inside GetSkillRangeType) to ObjectMgr.cpp  (reverting some changes on that update) seems to be a hotfix for both your bug and another one caused by professions and riding skills max values being affected by level up as well:

``` 
        case SKILL_CATEGORY_SECONDARY:
        case SKILL_CATEGORY_PROFESSION:
            // not set skills for professions and racial abilities
            if (IsProfessionOrRidingSkill(rcEntry->SkillID))
                return SKILL_RANGE_RANK;
```

**Comment by Einkatzi** [comment]
Afaik, the tables ending in _dbc are only supposed to override data in the corresponding DBC file. That's why most of them are empty.
DBCStores.cpp seems to create that link between *_dbc table and DBC file, but I can't read C++ well enough to understand what happens then (i.e. where the data is actually read from file or mysql).

**Comment by Onidsouza** [comment]
@Einkatzi I see, I misunderstood their purpose then. In any case, I did a quick search for all references for sSkillTiersStore in the code and couldn't find where is the code to properly load values into it. Since the code given seems to catch professions and riding skills, it means that for some reason the command `sSkillTiersStore.LookupEntry(rcEntry->SkillTierID)` is returning false for them. My first guess would then be that it looks for the value on some empty data structure (inside the program in runtime), but I don't know how to confirm it.

**Comment by necropola** [comment]
@Onidsouza Is this PR #6672 actually needed or is this case already covered by PR #6631?

**Comment by Einkatzi** [comment]
@Onidsouza If I'm interpreting this correctly, DBCDatabaseLoader.cpp does the job of connecting to the DBC database and querying from it. It's also there that values from the corresponding _dbc in the SQL database override the default records from the DBC. Looks like the SQL db is queried, nothing happens if it returns 0 records (bc there's nothing to override), DBC content is read into an array, and the SQL query content (if present) overwrites parts of the array. Here it is:
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/shared/DataStores/DBCDatabaseLoader.cpp
Care to have a second look to confirm this?

I think sSkillTiersStore does contain the necessary data, but I'm a bloody beginner at even reading C++.

**Comment by necropola** [comment]
This issues is no longer valid. It was fixed in #6631 a few days ago.
The problem was that `SkillTiers.dbc` was not loaed and hence `sSkillTiersStore` was empty.
Closing this issue.

@Onidsouza You can close your PR #6672, since it's no longer needed. Thanks.
